javav2/example_code/iotsitewise/src/main/java/com/example/iotsitewise/scenario/SitewiseScenario.java

```diff
@@ -0,0 +1,438 @@
+            to be secure and compliant, with features like role-based access controls, data encryption, 
+            and integration with other AWS services for additional security and compliance features.
+                        
+            Let's get started...
+            """);
+
+        waitForInputToContinue(scanner);
+        logger.info(DASHES);
+
+        try {
+            runScenario(assetModelName, assetName, portalName, contactEmail, gatewayName, myThing);
+        } catch (RuntimeException e) {
+           logger.info(e.getMessage());
+        }
+    }
+
+    public static void runScenario(String assetModelName, String assetName,  String portalName, String contactEmail, String gatewayName, String myThing) throws Throwable {
+        logger.info("Use AWS CloudFormation to create an IAM role that is required for this scenario.");
+        CloudFormationHelper.deployCloudFormationStack(ROLES_STACK);
+        Map<String, String> stackOutputs = CloudFormationHelper.getStackOutputs(ROLES_STACK);
+        String iamRole = stackOutputs.get("SitewiseRoleArn");
+        logger.info("The ARN of the IAM role is {}",iamRole);
+        logger.info(DASHES);
+
+        logger.info(DASHES);
+        logger.info("1. Create an AWS SiteWise Asset Model");
+        logger.info("""
+             An AWS IoT SiteWise Asset Model is a way to represent the physical assets, such as equipment,
+             processes, and systems, that exist in an industrial environment. This model provides a structured and
+             hierarchical representation of these assets, allowing users to define the relationships and properties
+             of each asset.
+             
+             This scenario creates two asset model properties: temperature and humidity.
+            """);
+        waitForInputToContinue(scanner);
+        String assetModelId;
+        try {
+            CompletableFuture<CreateAssetModelResponse> future = sitewiseActions.createAssetModelAsync(assetModelName);
+            CreateAssetModelResponse response = future.join();
+            assetModelId = response.assetModelId();
+            logger.info("Asset Model successfully created. Asset Model ID: {}. ", assetModelId);
+
+        } catch (RuntimeException rt) {
+            Throwable cause = rt;
+            while (cause.getCause() != null && !(cause instanceof ResourceAlreadyExistsException)) {
+                cause = cause.getCause();
+            }
+            if (cause instanceof ResourceAlreadyExistsException) {
+                assetModelId = sitewiseActions.getAssetModelIdAsync(assetModelName).join();
+                logger.info("The Asset Model {} already exists. The id of the existing model is {}. Moving on...", assetModelName, assetModelId);
+            } else {
+                logger.info("An unexpected error occurred: " + rt.getMessage(), rt);
+                throw cause;
```

Comment:
should this be a `return` rather than a throw?

Reply:
Thrtow back to main()